pandas/tests/io/test_compression.py

```diff
@@ -201,9 +201,13 @@ def test_gzip_reproducibility_file_object():
     assert output == buffer.getvalue()
 
 
-def test_with_missing_lzma():
+def test_with_missing_lzma(monkeypatch):
     """Tests if import pandas works when lzma is not present."""
     # https://github.com/pandas-dev/pandas/issues/27575
+    # Can't import pandas from the test directory since its not
+    # built inplace with meson
+    if pd._built_with_meson:
```

Comment:
Do we ever give users the opportunity to create a build folder wherever they want? A bit esoteric but guessing this would fail in cases where the user picks a sub-directory as their build folder

Reply:
`pip` / `build` can pass backend-defined settings, and the meson-python backend has a setting to choose the build folder. It's not particularly likely to be used.

The generalized solution to the whole topic (for any build backend, really) is https://peps.python.org/pep-0660/, or at least the infrastructure it requires, which enables "proper" importing of the build directory during development/testing.

There's an open ticket for meson-python to implement this.

Reply:
Hmm possibly. I think the limitation of that though is that you would still have to install, where maybe you wanted to have separate debug / release build folders that you could pull from

Like I said pretty estoric, so not something we need to spend a lot of time on. Just makes for interesting conversation

Reply:
"or at least the infrastructure it requires"

My thought is that running a testsuite could probably be made to hook into whatever mechanism a PEP 660 implementation uses, although without actually installing a wheel. For example, if the build backend generates the contents of an editable wheel as an internal build artifact with a documented path, that directory could be added to PYTHONPATH by the test runner.

It's mostly shower thoughts at the moment, but in the long run I'd want Meson to be able to do this via https://mesonbuild.com/Commands.html#devenv as well. You'd be able to run `meson devenv -C builddir/` and that would spawn a temporary shell where `python3` is hooked up to the editable install.